gp-includes/misc.php

```diff
@@ -508,3 +508,45 @@ function gp_wp_profile_options_update( $user_id ) {
 		$current_perm->delete();
 	}
 }
+
+/**
+ * Returns a mutli-dimensional array with the sort by types, descriptions and SQL query for each.
+ *
+ * @since 2.1.0
+ *
+ * @return array An array of sort by field types.
+ */
+function gp_get_sort_by_fields() {
+	$sort_fields = array(
+		'original_date_added' => array(
+			'title'       => __( 'Date added (original)', 'glotpress' ),
+			'sql_sort_by' => 'o.date_added %s',
+		),
+		'translation_date_added' => array(
+			'title'       => __( 'Date added (translation)', 'glotpress' ),
+			'sql_sort_by' => 't.date_added %s',
+		),
+		'original' => array(
+			'title'       => __( 'Original string', 'glotpress' ),
+			'sql_sort_by' => 'o.singular %s',
+		),
+		'translation' => array(
+			'title'       => __( 'Translation', 'glotpress' ),
+			'sql_sort_by' => 't.translation_0 %s',
+		),
+		'priority' => array(
+			'title'       => __( 'Priority', 'glotpress' ),
+			'sql_sort_by' => 'o.priority %s, o.date_added DESC',
+		),
+		'references' => array(
+			'title'       => __( 'Filename in source', 'glotpress' ),
+			'sql_sort_by' => 'o.references',
+		),
+		'random' => array(
+			'title'       => __( 'Random', 'glotpress' ),
+			'sql_sort_by' => 'o.priority DESC, RAND()',
+		),
+	);
+
+	return apply_filters( 'gp_sort_by_fields', $sort_fields );
```

Comment:
Needs documentation, see https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/php/#4-hooks-actions-and-filters.